new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElkSample.java

```diff
@@ -0,0 +1,122 @@
+
+/**
+ * Verify that Elasticsearch collects data from WebLogic logs and
+ * stores them in its repository correctly.
+ */
+@DisplayName("Test to use Elasticsearch API to query WebLogic logs")
+@IntegrationTest
+class ItElkSample {
+
+  // constants for Operator
+  private static String opNamespace = null;
+  private static String domainNamespace = null;
+  private static ConditionFactory withStandardRetryPolicy = null;
+
+  // constants for ELK Stack
+  private static LoggingExporterParams elasticsearchParams = null;
+  private static LoggingExporterParams kibanaParams = null;
+  private static LoggingFacade logger = null;
+
+  /**
+   * Install Elasticsearch, Kibana and Operator, create a one cluster domain.
+   *
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void init(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+      .and().with().pollInterval(10, SECONDS)
+      .atMost(5, MINUTES).await();
+
+    // get a new unique opNamespace
+    logger.info("Assigning a unique namespace for Operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    // get a new unique domainNamespace
+    logger.info("Assigning a unique namespace for Domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domainNamespace = namespaces.get(1);
+
+    // install and verify Elasticsearch
+    logger.info("install and verify Elasticsearch");
+    elasticsearchParams = assertDoesNotThrow(() -> installAndVerifyElasticsearch(),
+            String.format("Failed to install Elasticsearch"));
+    assertTrue(elasticsearchParams != null, "Failed to install Elasticsearch");
+
+    // install and verify Kibana
+    logger.info("install and verify Kibana");
+    kibanaParams = assertDoesNotThrow(() -> installAndVerifyKibana(),
+        String.format("Failed to install Kibana"));
+    assertTrue(kibanaParams != null, "Failed to install Kibana");
+
+    // install and verify Operator
+    installAndVerifyOperator(opNamespace, opNamespace + "-sa", false, 0, true, domainNamespace);
+  }
+
+  /**
+   * Uninstall ELK Stack and delete domain custom resource.
+   */
+  @AfterAll
+  void tearDown() {
+    // uninstall ELK Stack
+    if (elasticsearchParams != null) {
+      logger.info("Uninstall Elasticsearch pod");
+      assertDoesNotThrow(() -> uninstallAndVerifyElasticsearch(elasticsearchParams),
+          "uninstallAndVerifyElasticsearch failed with ApiException");
+    }
+
+    if (kibanaParams != null) {
+      logger.info("Uninstall Elasticsearch pod");
+      assertDoesNotThrow(() -> uninstallAndVerifyKibana(kibanaParams),
+          "uninstallAndVerifyKibana failed with ApiException");
+    }
+  }
+
+  /**
```

Comment:
Can you please describe the usecase detail with step-by-step description. It will be helpful to triage the failures. How you are asserting that ELK is ready to use

Reply:
fixed